synth.py

```diff
@@ -6,44 +6,38 @@
 
 logging.basicConfig(level=logging.DEBUG)
 
-gapic = gcp.GAPICGenerator()
+gapic_micro = gcp.GAPICMicrogenerator()
 
-# tasks has two product names, and a poorly named artman yaml
-v1_admin_library = gapic.node_library(
-    "firestore-admin", "v1", config_path="/google/firestore/admin/artman_firestore_v1.yaml"
+v1_admin_library = gapic_micro.typescript_library(
+    "firestore-admin", "v1", proto_path="/google/firestore/admin/v1",
+    generator_args={'grpc-service-config': 'google/firestore/admin/v1/firestore_admin_grpc_service_config.json'}
 )
-v1beta1_library = gapic.node_library(
-    "firestore", "v1beta1", config_path="/google/firestore/artman_firestore.yaml"
+v1beta1_library = gapic_micro.typescript_library(
+    "firestore", "v1beta1", proto_path="/google/firestore/v1beta1",
+    generator_args={'grpc-service-config': 'google/firestore/v1beta1/firestore_grpc_service_config.json'}
 )
-v1_library = gapic.node_library(
-    "firestore", "v1", config_path="/google/firestore/artman_firestore_v1.yaml"
+v1_library = gapic_micro.typescript_library(
+    "firestore", "v1", proto_path="/google/firestore/v1",
+    generator_args={'grpc-service-config': 'google/firestore/v1/firestore_grpc_service_config.json'}
 )
 
 # skip index, protos, package.json, and README.md
-s.copy(v1_admin_library, "dev", excludes=["package.json", "README.md", "src/index.js", "src/v1/index.js"])
-s.copy(v1beta1_library, "dev", excludes=["package.json", "README.md", "src/index.js", "src/v1beta1/index.js"])
-s.copy(v1_library, "dev", excludes=["package.json", "README.md", "src/index.js", "src/v1/index.js"])
-
-# package.json is one level deeper since firestore's src/ is under dev/
-s.replace(
-    "dev/src/v1/firestore_admin_client.js", "../../package.json", "../../../package.json"
-)
-s.replace(
-    "dev/src/v1beta1/firestore_client.js", "../../package.json", "../../../package.json"
-)
-s.replace(
-    "dev/src/v1/firestore_client.js", "../../package.json", "../../../package.json"
-)
+s.copy(v1_admin_library, "dev", excludes=["package.json", "README.md", "src/index.ts", "src/v1/index.ts", 
+    "tsconfig.json", "tslint.json", "linkinator.config.json", "webpack.config.js"])
+s.copy(v1beta1_library, "dev", excludes=["package.json", "README.md", "src/index.ts", "src/v1beta1/index.ts",
+    "tsconfig.json", "tslint.json", "linkinator.config.json", "webpack.config.js"])
+s.copy(v1_library, "dev", excludes=["package.json", "README.md", "src/index.ts", "src/v1/index.ts",
+    "tsconfig.json", "tslint.json", "linkinator.config.json", "webpack.config.js"])
 
 # Fix dropping of google-cloud-resource-header
 # See: https://github.com/googleapis/nodejs-firestore/pull/375
 s.replace(
-    "dev/src/v1beta1/firestore_client.js",
+    "dev/src/v1beta1/firestore_client.ts",
     "return this\._innerApiCalls\.listen\(options\);",
     "return this._innerApiCalls.listen({}, options);",
```

Comment:
Is this still needed (or rather: does the original bug still exist in the TS generated client?). It would seem to be that this should now be caught by the type system.